ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/TopicStream.java

```diff
@@ -150,87 +258,128 @@ public Format getFormat() {
       @Override
       public Optional<Formatter> maybeGetFormatter(
           final String topicName,
-          final ConsumerRecord<String, Bytes> record,
-          final KafkaAvroDeserializer avroDeserializer,
-          final DateFormat dateFormat) {
+          final Bytes data,
+          final KafkaAvroDeserializer avroDeserializer
+      ) {
         try {
-          final JsonNode jsonNode = JsonMapper.INSTANCE.mapper.readTree(record.value().toString());
+          final JsonNode jsonNode = JsonMapper.INSTANCE.mapper.readTree(data.toString());
 
-          // If the JsonNode is not structured like 'key:value', then do not use JSON to print
-          // this value
-          if (!(jsonNode instanceof ObjectNode)) {
+          if (!(jsonNode instanceof ObjectNode) && !(jsonNode instanceof ArrayNode)) {
+            // Other valid JSON types, e.g. NumericNode, BooleanNode, etc
+            // are indistinguishable from single column delimited format:
```

Comment:
this comment is a little confusing - so what if they're indistinguishable? Are we returning `Optional.empty` because we'd _rather_ default to DELIMITED than "inline" JSON?

Reply:
If this stays in the next version I'll add more info. However, I don't think it will be.